src/nnet3/nnet-training.cc

```diff
@@ -62,20 +63,78 @@ void NnetTrainer::Train(const NnetExample &eg) {
                         &request);
   const NnetComputation *computation = compiler_.Compile(request);
 
-  NnetComputer computer(config_.compute_config, *computation,
+  if (config_.backstitch_training_scale > 0.0 &&
```

Comment:
My comments about the chain training code all apply to the regular nnet3 training code also.